cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -333,14 +362,15 @@ class device_row_comparator {
 
     template <typename Element,
               CUDF_ENABLE_IF(not cudf::is_relationally_comparable<Element, Element>() and
-                             not std::is_same_v<Element, cudf::struct_view>)>
+                             (not has_nested_columns or not cudf::is_nested<Element>()))>
     __device__ cuda::std::pair<weak_ordering, int> operator()(size_type const,
                                                               size_type const) const noexcept
     {
       CUDF_UNREACHABLE("Attempted to compare elements of uncomparable types.");
     }
 
-    template <typename Element, CUDF_ENABLE_IF(std::is_same_v<Element, cudf::struct_view>)>
+    template <typename Element,
+              CUDF_ENABLE_IF(has_nested_columns and std::is_same_v<Element, cudf::struct_view>)>
```

Comment:
So I think we can easily specialize this function with `not has_nested_columns` and adopt the old comparison approach for non-nested structs (flattening + old lex comparator). But this seems a bit ugly here so I'm not sure.

Reply:
Struct or list columns (whether or not they themselves contain struct or lists) are still nested. But you're right that we could, in theory, add another level of dispatch based on whether we have nested columns containing other nested columns and do something more complex if we really needed to. I agree that it's ugly, though, and it's not obvious where we would stop here. I'm going to resolve this particular thread and continue the discussion on the overall PR conversation by responding to your previous queries, so free free to add more thoughts there.